### PR TITLE
Validation request

### PR DESCRIPTION
**Pull request by cp-sumi-k**
- Implemented auth flow.
- Generate client secret
- Validate the user with auth-code, redirect-URI, and refresh-token
- Get user details from validation response

### CODE REVIEW

File: auth/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+```go
+
+req := auth.New("team-id", "client-id", "key-id", "secret-key")
+
+resp, err := req.ValidateCode("auth-code") 
+
+// or
+
+resp, err := req.ValidateCodeWithRedirectURI("auth-code", "redirect-uri")
```
Comment (nidhi-canopas) [inline auth/README.md:31]: It would be better to mention, which method should be used when.

File: auth/README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+// or
+
+resp, err := req.ValidateRefreshToken("refresh-token") 
+
+if err != nil {
+	log.Fatal(err.Error())
+}
+
+// get validated user
+user, err := resp.GetUser(resp.IDToken)
```
Comment (nidhi-canopas) [inline auth/README.md:42]: Is it validated user or verified or authorised user?

File: auth/README.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+email, err := resp.Email(resp.IDToken)
+
+if err != nil {
+	log.Fatal(err.Error())
+}
+
+log.Println(email)
+
+// get validated user status
+userStatus, err := resp.RealUserStatus(resp.IDToken)
```
Comment (nidhi-canopas) [inline auth/README.md:70]: It would be better to mention, status include which states.

File: auth/README.md (hunk 2 above)
Comment (cp-sumi-k) [reply auth/README.md:42]: We get users from validation, that's why it's validated. Let's call it simply `user`

**Review by nidhi-canopas — APPROVED** [review]

**Pull request merged**